tests/e2e/specs/add-edit-category.test.js

```diff
@@ -0,0 +1,176 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	visitAdminPage,
+	pressKeyWithModifier,
+	pressKeyTimes
+} from '@wordpress/e2e-test-utils'
+import { addQueryArgs } from '@wordpress/url';
+
+describe( 'Categories tests', () => {
+	const title = 'New Category';
+	const query = addQueryArgs( '', {
+		taxonomy: 'category',
+	} );
+
+	beforeEach( async () => {
+		/**
+		 * Delete all categories before anything
+		 * This is useful because after running more than one test
+		 * there could be existing categories
+		 */
+		await visitAdminPage( 'edit-tags.php', query );
+		await page.$( '#bulk-action-selector-top' );
+		await page.waitForSelector( '[id^=cb-select-all-]' );
+		await page.click( '[id^=cb-select-all-]' );
+		await page.select( '#bulk-action-selector-top', 'delete' );
+		await page.click( '#doaction' );
+
+		/**
+		 * Create a new category with the title 'New Category'
+		 */
+		await page.waitForSelector('#tag-name');
+		await page.focus( '#tag-name' );
+		await pressKeyWithModifier( 'primary', 'a' );
+		await page.type( '#tag-name', title );
+		await page.click( '#submit' );
+
+		await page.reload();
+	} );
+
+	it( 'shows the new created category with the correct title', async () => {
+		// Expect there to be two rows in the categories list.
+		// The new created category and the default Uncategorized"
+		await page.waitForSelector( '#the-list tr' );
+		const categories = await page.$$( '#the-list tr' );
+		expect( categories.length ).toBe( 2 );
+
+		// Expect the new created category title to be correct.
+		const firstRowCategoryTitle = await page.$x(
+			`//a[contains(@class, "row-title")][contains(text(), "${ title }")]`
+		);
+		expect( firstRowCategoryTitle.length ).toBe( 1 );
+	} );
+
+	it( 'allows an existing category to be edited using the Edit button', async () => {
+		await page.waitForSelector( '#the-list tr' );
+		// Click the first (new created) category title (edit) link
+		const [ editLink ] = await page.$x(
+			`//a[contains(@class, "row-title")][contains(text(), "${ title }")]`
+		);
+		await editLink.click();
+
+		await page.waitForNavigation();
+
+		await page.waitForSelector( '.term-name-wrap input#name' );
+		// Expect to now be in the category editor with the correct category title shown.
+		const editorCategoryTitleInput = await page.$x(
+			`//input[contains(@id, "name")][@value = "${ title }"]`
+		);
+		expect( editorCategoryTitleInput.length ).toBe( 1 );
```

Comment:
Would be good to also edit the title, save and check that the list now displays the new title.

Reply:
I can reproduce the failure of the commented out test below locally, and I'm also getting a failure on `allows an existing category to be edited using the Edit button` in interactive mode.

For the above test failing in interactive mode, replacing `await editLink.click();` with `await editLink.focus();` followed by `await page.keyboard.press( 'Enter' );` fixes it for me.

For the below commented out test, it's failing right at the start, on `await page.waitForSelector( '#the-list tr' );`, and I can fix it locally by running `await deleteAllCategories();` and `await createNewCategory( categoryTitle );` beforehand. I'm not 100% sure what state the interface is in to cause that failure, but it's good practice to reset the environment before running each test, because we should be able to reliably run only a single test from a suite, or to skip some of the tests. In order to do that, no test should depend on mutations happening in previous tests.

So my suggestion is to add a `beforeEach` and run `deleteAllCategories` in it; then run `createNewCategory` at the start of any test that requires it.

It would also be good to consolidate these two tests into one (when I initially suggested editing and saving the title, that's what I had in mind). This because the test should do exactly what its description says, and `allows an existing category to be edited using the Edit button` implies that we are checking that the category can be successfully edited. Currently, the test only checks that we are able to load the editing page, but not that editing is actually possible. And because with e2e we generally attempt to test whole flows, it makes sense to consolidate this in one test instead of having one test that checks the page can be loaded and another that checks editing.